src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -11,207 +12,171 @@ namespace System.Buffers
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal bool TryGetBuffer(SequencePosition start, SequencePosition end, out ReadOnlyMemory<T> data, out SequencePosition next)
```

Comment:
Should the SequencePosition be `in`?

Reply:
Well... probably yes, but probably not on public api. At some point in the future the Jit may handle passing fields it passes by ref anyway (ByValRef?) without coping them first https://github.com/dotnet/corefx/issues/27238 and maybe decompose a two field 16byte struct into two registers on Unix.

Though it doesn't currently do this. So... `in` is probably good only non-public apis so they can be changed if the Jit did do this...

Don't let it stop you merging https://github.com/aspnet/KestrelHttpServer/pull/2295 and https://github.com/aspnet/Common/pull/311 😉

Reply:
This is internal 😄

Reply:
/cc @AndyAyersMS

Reply:
Ah I see the advice on that thread

Reply:
They wouldn't let me change every method in coreclr and corefx that took a `ReadOnlySpan`\`ReadOnlyMemory` 😢 😺

Reply:
Also changed `ReadOnlySequence<T> sequence` to `in ReadOnlySequence<T> sequence`